go/vt/vtgate/scatter_conn.go

```diff
@@ -660,15 +667,28 @@ func (stc *ScatterConn) multiGoTransaction(
 			oneShard(rs, i)
 		}
 	} else {
+		var panicRecord *panicData
 		var wg sync.WaitGroup
 		for i, rs := range rss {
 			wg.Add(1)
 			go func(rs *srvtopo.ResolvedShard, i int) {
 				defer wg.Done()
+				defer func() {
+					if r := recover(); r != nil {
+						panicRecord = &panicData{
```

Comment:
Since any goroutine can panic during parallel execution, it's possible for those panics to step on each other and trigger a data race writing to this field.

The easy solution is to add:
```go
panicMu.Lock()
defer panicMu.Unlock()
panicRecord = /* ... */
```

Alternatively you could create `var panics []panicData` and capture more than one vs overwriting any prior data with the last panic.

Reply:
I still do not see this as an issue, as there is no reading happening before all the go routines are complete. 
When the `read` happens there is no live goroutine trying to write to it. 

The code read one of the `panic` address references, which is the requirement of the issue.

Reply:
We run our unit test with `-race`. The new unit test added should fail if that would have been the case.

Reply:
Multiple goroutines can panic concurrently meaning that the race is on the writing side.

Reply:
Unfortunately running with `-race` does not guarantee that all races are detected, only races which are triggered during test execution. To trigger this you'd have to have multiple goroutines panic concurrently in a test.

Reply:
made the required changes.